src/lib.rs

```diff
@@ -110,56 +119,98 @@ impl<T: Config + Send + Sync> GgrsPlugin<T> {
     }
 
     /// Registers a type of resource for saving and loading during rollbacks.
-    pub fn register_rollback_resource<Type>(self) -> Self
+    pub fn register_rollback_resource<Type>(&mut self) -> &mut Self
     where
         Type: GetTypeRegistration + Reflect + Default + Resource,
     {
-        let mut registry = self.type_registry.write();
+        let mut registry = self.0.write();
         registry.register::<Type>();
 
         let registration = registry.get_mut(std::any::TypeId::of::<Type>()).unwrap();
         registration.insert(<ReflectResource as FromType<Type>>::from_type());
         drop(registry);
         self
     }
+}
 
-    /// Consumes the builder and makes changes on the bevy app according to the settings.
-    pub fn build(self, app: &mut App) {
-        let mut input_system = self
-            .input_system
-            .expect("Adding an input system through GGRSBuilder::with_input_system is required");
-        // ggrs stage
-        input_system.initialize(&mut app.world);
-        let mut stage = GgrsStage::<T>::new(input_system);
-        stage.set_update_frequency(self.fps);
+// Label for the schedule which reads the inputs for the current frame
+#[derive(ScheduleLabel, Debug, Hash, PartialEq, Eq, Clone)]
+pub struct ReadInputs;
+
+/// GGRS plugin for bevy.
+pub struct GgrsPlugin<C: Config> {
+    /// phantom marker for ggrs config
+    _marker: PhantomData<C>,
```

Comment:
Is this still the correct way to do this? In the past, I have seen this solution often, but it seems so clunky.

Reply:
Yeah, I don't think there's any way around it, unfortunately.